src/unit_tests/wazuh_modules/agent_upgrade/CMakeLists.txt

```diff
@@ -24,6 +24,9 @@ set_target_properties(
 target_link_libraries(UPGRADE_O ${WAZUHLIB} ${WAZUHEXT} -lpthread)
 
 # Generate agent upgrade tests
+list(APPEND upgrade_names "test_wm_agent_upgrade")
+list(APPEND upgrade_flags "-Wl,--wrap,_mtinfo -Wl,--wrap,pthread_exit -Wl,--wrap,wm_agent_upgrade_check_status -Wl,--wrap,wm_agent_upgrade_listen_messages")
+
 list(APPEND upgrade_names "test_wm_agent_upgrade_agent")
```

Comment:
You could avoid compiling some of the test files that do not correspond to a certain target using here lines such as:

```
if(${TARGET} STREQUAL "agent")
# Linux agent
elseif(${TARGET} STREQUAL "winagent")
# Win agent
else()
# Linux server
endif()
```

Instead of adding ifdefs inside the test file. Thay way you will avoid files that do not run any test at all.

Reply:
Done